hasher-matcher-actioner/hmalib/lambdas/api/lcc.py

```diff
@@ -22,13 +25,24 @@ def to_json(self) -> t.Dict:
         }
 
 
-def get_lcc_api() -> bottle.Bottle:
+@functools.lru_cache(maxsize=None)
+def get_index(storage_path, signal_type):
+    index = LCCIndexer.get_recent_index(storage_path, signal_type)
+    return index
+
+
+def get_lcc_api(storage_path) -> bottle.Bottle:
 
     lcc_api = SubApp()
 
     @lcc_api.get("/", apply=[jsoninator])
     def lcc_hasher() -> LCCResponse:
-        print("Hash: ", bottle.request.query.hash)
-        return LCCResponse(False, "", "")
+        index = get_index(storage_path, bottle.request.query.hash.signal_type)
+        hash_value = bottle.request.query.hash
+        match_array = index.query(hash_value)
+        match_value = False
+        if len(match_array) != 0:
+            match_value = True
+        return LCCResponse(match_value, hash_value, match_value[0].metadata)
```

Comment:
Add a todo suggesting that we might need to convert match_value[0].metadata into a preview url.